lib/extensions/permissions.js

```diff
@@ -103,24 +107,40 @@ async function setAccess (udid, bundleId, permissionsMapping) {
  * @param {string} serviceName - the name of the service. Should be one of
  * `SERVICES` keys.
  * @param {string} simDataRoot - the path to Simulator `data` root
- * @returns {string} - The current status: yes/no/unset
+ * @returns {string} - The current status: yes/no/unset/limited
  * @throws {Error} If there was an error while retrieving permissions.
  */
 async function getAccess (bundleId, serviceName, simDataRoot) {
   const internalServiceName = toInternalServiceName(serviceName);
   const dbPath = path.resolve(simDataRoot, 'Library', 'TCC', 'TCC.db');
-  for (const [sqlValue, status] of [['0', STATUS_NO], ['1', STATUS_YES]]) {
-    const sql = `SELECT count(*) FROM 'access' ` +
-      `WHERE client='${bundleId}' AND allowed=${sqlValue} AND service='${internalServiceName}'`;
-    const count = await execSQLiteQuery(dbPath, sql);
-    if (parseInt(count.split('=')[1], 10) > 0) {
-      return status;
+  const getAccessStatus = async (statusPairs, statusKey) => {
+    for (const [statusValue, status] of statusPairs) {
+      const sql = `SELECT count(*) FROM 'access' ` +
+        `WHERE client='${bundleId}' AND ${statusKey}=${statusValue} AND service='${internalServiceName}'`;
+      const count = await execSQLiteQuery(dbPath, sql);
+      if (parseInt(count.split('=')[1], 10) > 0) {
+        return status;
+      }
     }
+    return STATUS.UNSET;
+  };
+
+  // 'auth_value' existence depends on the OS version rather than Xcode version.
+  // Thus here check the newer one first, then fallback to the older version way.
+  try {
+    // iOS 14+
+    return await getAccessStatus(
+      [['0', STATUS.NO], ['2', STATUS.YES], ['3', STATUS.LIMITED]],
```

Comment:
what does value of '1' mean?

Reply:
Tried to find, but I was not able to figure `1` out yet. At least it is not in `YES|NO|limited|unset`

Reply:
`location=always|inuse|never|unset` did not match as well